File: Python_solutions/breadthFirstSearch.py
class Node:
    def __init__(self, name):
        self.children = []
        self.name = name

    def addChild(self, name):
        self.children.append(Node(name))
        return self

    def breadthFirstSearch(self, array):
        que = [self]
        while len(que):
            curr = que.pop(0)
            array.append(curr.name)
            que.extend(curr.children)
        return array

File: easy-problems/js-solutions/breadthFirstSearch.js
class Node {
  constructor(name) {
    this.name = name;
    this.children = [];
  }

  addChild(name) {
    this.children.push(new Node(name));
    return this;
  }

  breadthFirstSearch(array) {
    const que = [this];
    while (que.length) {
      let { name, children } = que.shift();
      array.push(name);
      if (children.length) que.push(...children);
    }
    return array;
  }
}
